### PR TITLE
Feature 1684: Add ArkiveExtension to disable preview parameters

### PR DESCRIPTION
**Pull request by spasepremcheski**
## Summary

Add support for disabling Preview parameters

### Type

- [x] **Feature**: This pull request introduces a new feature.
- [ ] **Bug fix**: This pull request fixes a bug.
- [ ] **Refactor**: This pull request refactors existing code.
- [ ] **Documentation**: This pull request updates documentation.
- [ ] **Other**: This pull request makes other changes.

#### Additional information

- [ ] This pull request introduces a **breaking change**.

### Description

- Add `disablePreviewParameters` to ArkiveExtension
- Pass it to ComposeVariantSpec via KSP arguments
- Add example in sample app

## Checklist

- [x] I have performed a self-review of my own code.
- [x] I have tested my changes, including edge cases.
- [ ] I have added necessary tests for the changes introduced (if applicable).
- [ ] I have updated the documentation to reflect my changes (if applicable).

### CODE REVIEW

File: plugin/src/main/java/com/infinum/arkive/plugin/ArkivePlugin.kt [hunk 1]
```diff
@@ -27,10 +28,25 @@ class ArkivePlugin : Plugin<Project> {
...
+                val disablePreviewParameters = extension.disablePreviewParameters.get()
+
+                project.extensions.findByName("ksp")?.let { kspExt ->
+                    try {
+                        val argMethod = kspExt.javaClass.getMethod("arg", String::class.java, String::class.java)
+                        argMethod.invoke(kspExt, DISABLE_PREVIEW_PARAMETERS, disablePreviewParameters.toString())
+                    } catch (e: Exception) {
+                        project.logger.warn("Failed to pass $DISABLE_PREVIEW_PARAMETERS to KSP: ${e.message}")
+                    }
+                }
```
Comment (spasepremcheski) [inline plugin/src/main/java/com/infinum/arkive/plugin/ArkivePlugin.kt:48]: @thisAAY please check this approach of passing the argument to KSP args and retrieving it in `ArkiveProcessorProvider`

**Review by thisAAY — APPROVED** [review]
